src/unix/linux-core.c

```diff
@@ -932,3 +932,72 @@ void uv__set_process_title(const char* title) {
   prctl(PR_SET_NAME, title);  /* Only copies first 16 characters. */
 #endif
 }
+
+
+static uint64_t uv__read_proc_meminfo(const char* what) {
+  unsigned long rc;
+  ssize_t n;
+  char* p;
+  int fd;
+  char buf[4096];  /* Large enough to hold all of /proc/meminfo. */
+
+  rc = 0;
+  fd = uv__open_cloexec("/proc/meminfo", O_RDONLY);
+
+  if (fd == -1)
+    return 0;
+
+  n = read(fd, buf, sizeof(buf) - 1);
+
+  if (n <= 0)
+    goto out;
+
+  buf[n] = '\0';
+  p = strstr(buf, what);
+
+  if (p == NULL)
+    goto out;
+
+  p += strlen(what);
+
+  if (1 != sscanf(p, "%lu kB", &rc))
+    goto out;
+
+  rc *= 1024;
+
+out:
+
+  if (uv__close_nocheckstdio(fd))
+    abort();
+
+  return rc;
+}
+
+
+static uint64_t uv__get_memory(const char* what, unsigned offset) {
```

Comment:
Should type of `offset` be `size_t` to match the documented type of `offsetof()` macro? Otherwise some compilers/code analyzers may complain of lost precision...

Reply:
`offsetof()` evaluates to a compile-time constant so I don't think there's a situation where the compiler can't deduce that it's in range but I changed it to `size_t` anyway. :-)